### ISSUE TITLE
Deployment rolling update `maxSurge` and `maxUnavailable` not changed if I don't specify them

### ISSUE THREAD

**Issue by wuzhuoquan** [kubernetes/kubernetes#118350]
### What happened?

I have and exists deployment，the first time I apply It with specifying the rollingUpdate like this
```
  strategy:
    rollingUpdate:
      maxSurge: 100%
      maxUnavailable: 20%
    type: RollingUpdate
```

And then I modify the deployment's yaml，remove the strategy Setting，according to the default，I think after re-apply the deployment，the maxSurge and maxUnavailable will change to 25%，but after apply I get the deployment with "-o yaml " option，they do not change. Still like this:
```
  strategy:
    rollingUpdate:
      maxSurge: 100%
      maxUnavailable: 20%
    type: RollingUpdate
```

### What did you expect to happen?

After modify the  yaml and remove the strategy，and apply it, the deployment in cluster should change the maxSurge and maxUnavailable to 25% according to default.

### How can we reproduce it (as minimally and precisely as possible)?

Deploy a deployment with strategy like this
```
  strategy:
    rollingUpdate:
      maxSurge: 100%
      maxUnavailable: 20%
    type: RollingUpdate
```
then remove the strategy setting from the yaml and apply again

### Kubernetes version

<details>

```
Client Version: version.Info{Major:"1", Minor:"24", GitVersion:"v1.24.6", GitCommit:"b39bf148cd654599a52e867485c02c4f9d28b312", GitTreeState:"clean", BuildDate:"2022-09-21T13:19:24Z", GoVersion:"go1.18.6", Compiler:"gc", Platform:"darwin/amd64"}
Kustomize Version: v4.5.4
Server Version: version.Info{Major:"1", Minor:"24+", GitVersion:"v1.24.6-aliyun.1", GitCommit:"5296768e052ba56e92b5d5bf7b52d33973a34c6f", GitTreeState:"clean", BuildDate:"2023-04-19T06:36:28Z", GoVersion:"go1.18.6", Compiler:"gc", Platform:"linux/amd64"}
```

</details>

### Cloud provider

<details>
ali cloud
</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

**Comment by tamilselvan1102** [comment]
> kubectl rollout restart -f deploy.yaml --selector=component=deploy

Add additional label of **component: deploy**. This will update all additional config

**Comment by sftim** [comment]
/retitle Deployment rolling update `maxSurge` and `maxUnavailable` not changed if I don't specify them

**Comment by sftim** [comment]
I think this might be a partly due to a misunderstanding about how `kubectl apply` works.

**Comment by wuzhuoquan** [comment]
> 



> > kubectl rollout restart -f deploy.yaml --selector=component=deploy
> 
> Add additional label of **component: deploy**. This will update all additional config

I have try to apply the yaml file with --selector option, but the ```maxSurge``` and ```maxUnavailable``` still not change.

May be you misunderstand my problem.

My question is , if i don't specify the ```strategy``` in the yaml file ,then the ```maxSurge``` and ```maxUnavailable``` in the running Deployment in cluster will not change by default, the first time I specify ```maxSurge: 1``` and ```maxUnavailable: 1```，if i want to change these two values to default 25%, I must  specify them in the yaml explicitly, why?

**Comment by tamilselvan1102** [comment]
In your case the default strategy was overridden by the first deployment. if you want to update, have to mention the strategy again else it will consider last updated strategy.

**Comment by mochizuki875** [comment]
It seems to have been fixed at least v1.27 .

```console
$ kubectl version --short
Flag --short has been deprecated, and will be removed in the future. The --short output will become the default.
Client Version: v1.27.3
Kustomize Version: v5.0.1
Server Version: v1.27.3
```

Create `Deployment` using this manifest and check `.spec.strategy` field.


<details>

`nginx-deployment.yaml `
```yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: nginx-deployment
  labels:
    app: nginx
spec:
  replicas: 3
  selector:
    matchLabels:
      app: nginx
  strategy:
    rollingUpdate:
      maxSurge: 100%
      maxUnavailable: 20%
    type: RollingUpdate
  template:
    metadata:
      labels:
        app: nginx
    spec:
      containers:
      - name: nginx
        image: nginx
        ports:
        - containerPort: 80
```

</details>


```console
$ kubectl apply -f nginx-deployment.yaml
deployment.apps/nginx-deployment created

$ kubectl get deploy nginx-deployment -o jsonpath='{.spec.strategy}' 
{"rollingUpdate":{"maxSurge":"100%","maxUnavailable":"20%"},"type":"RollingUpdate"}
```

Next, remove `.spec.strategy` field and apply it again.
`.spec.strategy` field has been changed.

<details>

`nginx-deployment.yaml `
```yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: nginx-deployment
  labels:
    app: nginx
spec:
  replicas: 3
  selector:
    matchLabels:
      app: nginx
  # strategy:
  #   rollingUpdate:
  #     maxSurge: 100%
  #     maxUnavailable: 20%
  #   type: RollingUpdate
  template:
    metadata:
      labels:
        app: nginx
    spec:
      containers:
      - name: nginx
        image: nginx
        ports:
        - containerPort: 80
```

</details>


```console
$ kubectl apply -f nginx-deployment.yaml
deployment.apps/nginx-deployment configured

$ kubectl get deploy nginx-deployment -o jsonpath='{.spec.strategy}'
{"rollingUpdate":{"maxSurge":"25%","maxUnavailable":"25%"},"type":"RollingUpdate"}
```

**Comment by wuzhuoquan** [comment]
> It seems to have been fixed at least v1.27 .
> 
> ```
> $ kubectl version --short
> Flag --short has been deprecated, and will be removed in the future. The --short output will become the default.
> Client Version: v1.27.3
> Kustomize Version: v5.0.1
> Server Version: v1.27.3
> ```
> 
> Create `Deployment` using this manifest and check `.spec.strategy` field.
> 
> ```
> $ kubectl apply -f nginx-deployment.yaml
> deployment.apps/nginx-deployment created
> 
> $ kubectl get deploy nginx-deployment -o jsonpath='{.spec.strategy}' 
> {"rollingUpdate":{"maxSurge":"100%","maxUnavailable":"20%"},"type":"RollingUpdate"}
> ```
> 
> Next, remove `.spec.strategy` field and apply it again. `.spec.strategy` field has been changed.
> 
> ```
> $ kubectl apply -f nginx-deployment.yaml
> deployment.apps/nginx-deployment configured
> 
> $ kubectl get deploy nginx-deployment -o jsonpath='{.spec.strategy}'
> {"rollingUpdate":{"maxSurge":"25%","maxUnavailable":"25%"},"type":"RollingUpdate"}
> ```

thanks for your answer，I will try it later.

**Comment by tamilselvan1102** [comment]
@wuzhuoquan 
if you got a answer, please close the issue
